lib/view/seasons/seasons_screen.dart

```diff
@@ -73,18 +74,18 @@ class _SeasonsPageState extends State<SeasonsPage> {
   InkWell listTitleItem(
       Season model, int index, String seasonDocId, BuildContext context) {
     return InkWell(
-      onTap: () =>onTapItem(model,seasonDocId),
-      child: containerOfItem(index, model),
+      onTap: () => onTapItem(model, seasonDocId),
+      child: PrimaryContainer(
+          index: index,
+          rightTopText: model.year,
+          rightBottomText: model.seasonType),
     );
   }
-  onTapItem(Season model, String seasonDocId){
+
+  onTapItem(Season model, String seasonDocId) {
     //to clear the previous data in the next pages
-    context
-        .read<EventsLogic>()
-        .preparingSpecificEvents(model.eventsDocIds);
-    context
-        .read<StudentsLogic>()
-        .preparingSpecificStudents(studentsDocIds: model.studentsDocIds);
+    context.read<EventsLogic>()..specificEventsCleared();
```

Comment:
Are you sure that you need to do all of these "context.read"?

Reply:
solved #14